### PR TITLE
Added Custom IAM roles For STS

### PR DESCRIPTION
**Pull request by pvasant**

### CODE REVIEW

**Comment by pvasant** [comment]
@vkareh  PTAL

File: model/clusters_mgmt/v1/sts_type.model [hunk 1]
```diff
@@ -27,4 +27,10 @@ struct STS {
 
 	// List of roles necessary to access the AWS resources of the various operators used during installation
 	OperatorIAMRoles []OperatorIAMRole
+
+    //Custom IAM roles
+    CustomIAMRoles CustomIAMRoles
```
Comment (vkareh) [inline model/clusters_mgmt/v1/sts_type.model:32]: Indentation looks off here

File: model/clusters_mgmt/v1/custom_iam_roles.model [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Contains the necessary attributes to support role-based authentication on AWS.
+struct CustomIAMRoles {
+	// Master IAMRole
+	MasterIAMRole String
+
+	// Worker IAMRole
```
Comment (vkareh) [inline model/clusters_mgmt/v1/custom_iam_roles.model:22]: I would update these comments, as they show up in the API spec, to:
* The name of the IAM role that will be attached to master instances
* The name of the IAM role that will be attached to worker instances

File: model/clusters_mgmt/v1/sts_type.model [hunk 3]
```diff
@@ -27,4 +27,10 @@ struct STS {
 
 	// List of roles necessary to access the AWS resources of the various operators used during installation
 	OperatorIAMRoles []OperatorIAMRole
+
+    //Custom IAM roles
```
Comment (vkareh) [inline model/clusters_mgmt/v1/sts_type.model:31]: I would change the comment here to:
* Custom IAM roles to use for the instance profiles of the master and worker instances

**Comment by pvasant** [comment]
@vkareh PTAL  Addressed the comments

**Pull request merged**